comp/dogstatsd/server/server.go

```diff
@@ -371,12 +373,14 @@ func (s *server) start(context.Context) error {
 	}
 
 	if len(socketPath) > 0 {
-		unixListener, err := listeners.NewUDSDatagramListener(packetsChannel, sharedPacketPoolManager, sharedUDSOobPoolManager, s.config, s.tCapture, s.wmeta, s.pidMap, s.listernersTelemetry, s.packetsTelemetry, s.telemetry)
-		if err != nil {
-			s.log.Errorf("Can't init listener: %s", err.Error())
-		} else {
-			tmpListeners = append(tmpListeners, unixListener)
-			udsListenerRunning = true
+		if _, err := os.Stat(filepath.Dir(socketPath)); !os.IsNotExist(err) {
```

Comment:
```suggestion
		if _, err := os.Stat(filepath.Dir(socketPath)); !errors.Is(err, fs.ErrNotExist) {
```

Reply:
Also, should we check if the socket is not closed already? 

```go
if _, err := os.Stat(filepath.Dir(socketPath)); !errors.Is(err, fs.ErrNotExist) && !errors.Is(err, fs. ErrClosed)
```

Reply:
`Stat` on a directory probably won't ever return `ErrClosed`.

@ichinaski Do you think we can remove this entire condition and restore logic the way it was originally? We should always report something if we couldn't create the socket.